vpr/src/place/place.cpp

```diff
@@ -192,454 +89,26 @@ void try_place(const Netlist<>& net_list,
-    const auto& p_timing_ctx = placer_state.timing();
-    const auto& p_runtime_ctx = placer_state.runtime();
-
-    vtr::RngContainer rng(placer_opts.seed);
+    auto& place_ctx = g_vpr_ctx.mutable_placement();
+    place_ctx.lock_loc_vars();
+    place_ctx.compressed_block_grids = create_compressed_block_grids();
 
-    std::optional<NocCostHandler> noc_cost_handler;
-    // create cost handler objects
-    NetCostHandler net_cost_handler = alloc_and_load_placement_structs(placer_opts, noc_opts, directs,
-                                                                       placer_state, noc_cost_handler);
+    Placer placer(net_list, placer_opts, analysis_opts, noc_opts, directs, place_delay_model, cube_bb, is_flat, /*quiet=*/false);
 
 #ifndef NO_GRAPHICS
-    if (noc_cost_handler.has_value()) {
-        get_draw_state_vars()->set_noc_link_bandwidth_usages_ref(noc_cost_handler->get_link_bandwidth_usages());
+    if (placer.noc_cost_handler().has_value()) {
+        get_draw_state_vars()->set_noc_link_bandwidth_usages_ref(placer.noc_cost_handler()->get_link_bandwidth_usages());
     }
 #endif
 
-    vtr::ScopedStartFinishTimer timer("Placement");
-
-    if (noc_opts.noc) {
-        normalize_noc_cost_weighting_factor(const_cast<t_noc_opts&>(noc_opts));
-    }
-
-    initial_placement(placer_opts, placer_opts.constraints_file.c_str(),
-                      noc_opts, blk_loc_registry, noc_cost_handler, rng);
-
-    //create the move generator based on the chosen strategy
-    auto [move_generator, move_generator2] = create_move_generators(placer_state, placer_opts, move_lim, noc_opts.noc_centroid_weight, rng);
-
-    if (!placer_opts.write_initial_place_file.empty()) {
-        print_place(nullptr, nullptr, placer_opts.write_initial_place_file.c_str(), placer_state.block_locs());
-    }
-
-#ifdef ENABLE_ANALYTIC_PLACE
-    /*
-     * Analytic Placer:
-     *  Passes in the initial_placement via vpr_context, and passes its placement back via locations marked on
-     *  both the clb_netlist and the gird.
-     *  Most of anneal is disabled later by setting initial temperature to 0 and only further optimizes in quench
-     */
-    if (placer_opts.enable_analytic_placer) {
-        AnalyticPlacer{blk_loc_registry}.ap_place();
-    }
-
-#endif /* ENABLE_ANALYTIC_PLACE */
-
-    // Update physical pin values
-    for (const ClusterBlockId block_id : cluster_ctx.clb_nlist.blocks()) {
-        blk_loc_registry.place_sync_external_block_connections(block_id);
-    }
-
     const int width_fac = placer_opts.place_chan_width;
-    init_draw_coords((float)width_fac, blk_loc_registry);
-
-    /* Allocated here because it goes into timing critical code where each memory allocation is expensive */
-    IntraLbPbPinLookup pb_gpin_lookup(device_ctx.logical_block_types);
-    //Enables fast look-up of atom pins connect to CLB pins
-    ClusteredPinAtomPinsLookup netlist_pin_lookup(cluster_ctx.clb_nlist, atom_ctx.nlist, pb_gpin_lookup);
```

Comment:
@vaughnbetz 
`pb_gpin_lookup` and `netlist_pin_lookup` are initialized after the placement timer starts in line 215. Is their initialization times intentionally included in the total placement time?

Reply:
Probably just what was convenient. They can be moved outside placement if it cleans up the code, but please measure how long they take and if it can be significant they should get their own timers (either on their own or together).